### PR TITLE
add user key functionality

### PR DESCRIPTION
**Pull request by alexandru-uta**

### CODE REVIEW

File: backend/service.did [hunk 1]
```diff
@@ -1,7 +1,7 @@
 type file_id = nat64;
 
 type file_metadata = record {
-  file_id;
+  file_id: file_id;
```
Comment (ielashi) [inline backend/service.did:4]: nit: I believe this change is equivalent to what it used to be.

File: backend/service.did [hunk 2]
```diff
@@ -59,7 +60,7 @@ type upload_file_atomic_request = record {
 service docutrack : {
   hello_world : () -> (text);
 
-  set_user: (first_name: text, last_name: text) -> ();
+  set_user: (first_name: text, last_name: text, public_key: blob) -> ();
```
Comment (ielashi) [inline backend/service.did:63]: I suggest we add a user record for easier backward compatibility in the future.

```suggestion
  type user = record {
    first_name: text;
    last_name: text;
    public_key: blob;
  };
  
  set_user: (user) -> ();
```

File: backend/src/api/get_files.rs [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    match state.file_owners.get(&caller) {
+        None => vec![],
+        Some(file_ids) => file_ids
+            .iter()
+            .map(|file_id| PublicFileMetadata {
+                file_id: *file_id,
+                file_name: state
+                    .file_data
+                    .get(file_id)
+                    .unwrap()
```
Comment (ielashi) [inline backend/src/api/get_files.rs:14]: I'd add a message to communicate why this isn't expect to fail

```suggestion
                    .expect("file must exist")
```

File: backend/src/api/get_files.rs [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use crate::api::{request_file, set_user_info};
+    use ic_cdk::export::Principal;
+
+    #[test]
+    fn get_files_test() {
```
Comment (ielashi) [inline backend/src/api/get_files.rs:30]: nice test case :)

File: backend/src/api/user_info.rs [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+            first_name,
+            last_name,
+            public_key,
+        },
+    );
+}
+
+pub fn get_user_key(state: &State, caller: Principal) -> Vec<u8> {
+    // assumes the user data is already set
+    state.users.get(&caller).unwrap().public_key.clone()
```
Comment (ielashi) [inline backend/src/api/user_info.rs:23]: ditto
```suggestion
    state.users.get(&caller).expect("user does not exist.").public_key.clone()
```

**Review by ielashi — APPROVED** [review]

File: backend/service.did (hunk 1 above)
Comment (alexandru-uta) [reply backend/service.did:4]: for some reason candid got confused and returned an error without specifying a name here for the field.

**Pull request merged**